src/device_properties.cpp

```diff
@@ -124,7 +128,7 @@ struct cvk_device_properties_amd : public cvk_device_properties {
     }
     cl_uint get_max_first_cmd_batch_size() const override final { return 10; }
     cl_uint get_max_cmd_group_size() const override final { return 1; }
-    const std::set<std::string> get_native_builtins() const override final {
+    std::set<std::string> get_native_builtins() const override final {
```

Comment:
Technically a separate change but we could make the sets static and return a const reference to them instead of performance copies. I'm not too concerned about the performance impact given where it's called though.

Reply:
I'll remove that and make another PR for it later.